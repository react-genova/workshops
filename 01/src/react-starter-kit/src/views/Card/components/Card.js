//@flow
import * as React from 'react';
import styled from 'styled-components';

const Container = styled.div`
    border-radius: 0.5em;
    background-color: white;
    border: 0.035em solid #C8C8C8;
    font-family: Arial, Helvetica, sans-serif;
`;

const TitleContainer = styled.div`
    padding: 0.5em;
`;

const Title = styled.div`
    font-size: 1.6em;
    font-weight: bold;
    color: #000080;
`;

const SubTitle = styled.div`
    font-size: 1.3em;
    color: #808080;
`;

const Header = styled.div`
    border-bottom: 0.035em solid #C8C8C8;
`;

const Body = styled.div`
    padding: 0.5em;
`;

type Props = {
    title:string,
    description:string,
    children?:React.Node,
};
const Card = ({title, description, children}:Props) => (
    <Container>
        <Header>
            <TitleContainer>
                <Title>{title}</Title>
                <SubTitle>{description}</SubTitle>
            </TitleContainer>
        </Header>
        <Body>
            {children}
        </Body>
    </Container>
);

export default Card;