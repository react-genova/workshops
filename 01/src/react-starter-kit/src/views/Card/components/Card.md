### Card ###

```js
const onClick = ()=>null;
<Card title="Most popular sites">
    <CardElement title="e-commerce" description="A bunch of sites" url="" onClick={onClick}>
        <CardElement title="Amazon" description="West Side Store" url="https://www.amazon.it/" onClick={onClick}/>
        <CardElement title="Alibaba" description="East but not least" url="https://www.alibaba.com/" onClick={onClick}/>
    </CardElement>
    <CardElement title="Search engines" description="A not  exhaustive list of search engines" url="" onClick={onClick}>
        <CardElement title="Google" description="One of the best search engine" url="https://www.google.it/" onClick={onClick}/>
        <CardElement title="Duck Duck Go" description="Another search engine" url="https://duckduckgo.com/" onClick={onClick}/>
    </CardElement>
</Card>
```