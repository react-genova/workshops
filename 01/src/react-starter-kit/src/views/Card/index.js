//@flow
export {default as CardElement} from './components/CardElement'
export {default as Card} from './components/Card';
export {default} from './components/Card';