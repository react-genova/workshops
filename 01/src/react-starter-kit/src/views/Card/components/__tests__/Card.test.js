//@flow
import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card component', () => {

    it('Card snapshot', () => {
        const component = shallow(
            <Card title="I have 2 children" description="">
                <div>The first child</div>
                <div>The second child</div>
            </Card>
        );
        expect(component).toMatchSnapshot();
    });

});