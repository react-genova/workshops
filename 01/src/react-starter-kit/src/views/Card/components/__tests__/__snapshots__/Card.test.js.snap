// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        I have 2 children
      </styled.div>
      <styled.div />
    </styled.div>
  </styled.div>
  <styled.div>
    <div>
      The first child
    </div>
    <div>
      The second child
    </div>
  </styled.div>
</styled.div>
`;
